### PR TITLE
💻 Guest teacher connect option in teacher signup

### LINKED ISSUE #5428: 💻 Two new fields in the sign up page for teachers
In https://github.com/hedyorg/hedy/issues/5212 we added a tickmark for pairing, we need one more, for student/company support that we are going to roll out soon(ish):

[ ] I would like to be connected with a guest teacher who can give a few lessons (tickmark)

If this is ticked, we want the teacher's phone number too

### PR DESCRIPTION
**Pull request by Annelein**
Fixes #5428 

**How to test**
1. Go to the sign up page for teachers
2. Check the checkbox for connecting to a guest teacher
3. Make sure the phone number field pops up
4. Make sure only numbers are allowed in this field, not text.
Screenshot 2024-04-24 at 15 05 58

### CODE REVIEW

File: templates/signup.html [hunk 1]
```diff
@@ -44,6 +44,12 @@ <h4 class="text-blue-500 mt-0 mb-0 text-center">{{_('create_student_account_expl
                     <label class="signup-label" for="email">{{_('email')}}</label>
                     <input class="signup-input" id="email" name="email" type="email" required>
                 </div>
+                <div class="signup-container">
+                    <div class="hidden" id="phone_number">
+                        <label class="signup-label" for="phone">{{_('phone_number')}}</label>
+                        <input required class="signup-input" id="phone" name="phone" type="number" placeholder="{{_('landcode_phone_number')}}">
```
Comment (hasan-sh) [inline templates/signup.html:50]:
```suggestion
                        <input class="signup-input" id="phone" name="phone" type="number" placeholder="{{_('landcode_phone_number')}}">
```

File: templates/signup.html [hunk 2]
```diff
@@ -203,6 +209,11 @@ <h4 class="text-blue-500 mt-0 mb-0 text-center">{{_('create_student_account_expl
                             <label for="pair_with_teacher">{{_('pair_with_teacher')}}</label>
                             <input type="checkbox" id="pair_with_teacher" name="pair_with_teacher">
                         </div>
+                        <div class="flex justify-between items-center">
+                            <label for="connect_guest_teacher">{{_('connect_guest_teacher')}}</label>
+                            <input type="checkbox" id="connect_guest_teacher" name="connect_guest_teacher"
+                            _="on click toggle .hidden on #phone_number">
```
Comment (hasan-sh) [inline templates/signup.html:215]:
```suggestion
                            _="on click toggle [@required='true'] on #phone
                                     toggle .hidden on #phone_number">
```

**Review by hasan-sh — COMMENTED** [review]
This way you only make it required when it's in use!

File: templates/signup.html (hunk 2 above)
Comment (Annelein) [reply templates/signup.html:215]: Ah perfect, couldn't find how to do it in htmx! Thanks a lot

**Review by hasan-sh — APPROVED** [review]
Nothing to add!

**Pull request merged**